backend/tracim_backend/lib/core/user.py

```diff
@@ -243,9 +246,70 @@ def get_members_of_workspaces(self, workspace_ids: typing.List[int]) -> typing.L
         )
         return [item[0] for item in user_ids_in_workspaces_tuples]
 
+    def get_common_user_workspaces_for_user(self, user_id: int) -> List[Tuple[Any, ...]]:
+        """
+        Return list of UserRoleInWorkspace that are in common with the provided user
+        :param user_id: id of the user to get data from
+        :return: List of found UserRoleInWorkspace in common with the provided user
+        """
+        from sqlalchemy.orm import aliased
+
+        s1 = aliased(UserRoleInWorkspace)
+        s2 = aliased(UserRoleInWorkspace)
+
+        return (
+            self._session.query(s2.user_id, User.username, User.display_name, s2.workspace_id)
+            .join(s1, s1.workspace_id == s2.workspace_id)
+            .join(User, User.user_id == s2.user_id)
+            .filter(s1.user_id == user_id)
+            .order_by(s2.user_id)
+            .all()
+        )
+
+    def get_all_known_users_of_user(self, user_id: int) -> typing.List[User]:
+        """
+        Return list of known users of the provided users.
+        This list includes all users that share a same workspace with the user.
+        It also adds a list of shared workspaces per user to the user model.
+        :param user_id: user_id to get know_users of
+        :return: List of found users, with a list of shared workspaces per user
+        """
+
+        user_workspace_settings = self.get_common_user_workspaces_for_user(user_id)
+
+        if user_workspace_settings is None:
+            return []
+
+        # INFO- M.L. - 2023-11-23 - This loop relies on the above query
+        #  that returns an ordered list of users and their associated workspaces
+        #  with one entry per user/workspace relationship.
+        users = []
+        prev_user_id = None
+        # Initialises index to -1 to prevent initialisation snippet before the loop
+        current_index = -1
+        for user_workspace_setting in user_workspace_settings:
+            # Since the list is ordered, when the previous user_id is different
+            #  from the current one, initialise a new entry containing minimum user_info
+            #  and add the workspace of the current relationship into the list of the
+            #  user's workspaces
+            if prev_user_id is not user_workspace_setting[0]:
+                current_index += 1
+                user = User()
+                user.user_id = user_workspace_setting[0]
+                user.username = user_workspace_setting[1]
+                user.display_name = user_workspace_setting[2]
+                user.workspace_ids = [user_workspace_setting[3]]
+                users.append(user)
+                prev_user_id = user.user_id
+                continue
+            # If the user_id is the same as the precedent one, simply add the workspace
```

Comment:
avoid usage of words like "simply", "just", "easily" and such in documentations

Reply:
Removed in last commit

Reply:
fixed